support-workers/src/test/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionBuilderSpec.scala

```diff
@@ -112,9 +123,23 @@ class DigitalSubscriptionBuilderSpec extends AsyncFlatSpec with Matchers {
     UUID.fromString("f7651338-5d94-4f57-85fd-262030de9ad5"),
     SubscriptionPurchase(ZuoraDigitalPackConfig(14, 2), None, Quarterly, Country.UK, promotionService),
     SANDBOX,
+    giftCodeGeneratorService,
     () => saleDate
   ).value.map(_.right.get)
 
-  //TODO: RB test all cases
+  lazy val threeMonthGiftRedemption: Future[Either[PromoError, GetCodeStatus.RedemptionInvalid]] = DigitalSubscriptionBuilder.build(
+    DigitalPack(GBP, Quarterly, Gift),
+    UUID.fromString("f7651338-5d94-4f57-85fd-262030de9ad5"),
+    SubscriptionRedemption(RedemptionData(RedemptionCode("any-code").right.get),
+      new GetCodeStatus({
+        case "CODE" => Future.successful(Some(Map(
+          "available" -> DynamoLookup.DynamoBoolean(true),
+          "corporateId" -> DynamoLookup.DynamoString("1")
+        )))
+      })),
```

Comment:
I'm struggling to understand the types a bit here, is the indentation 2 too many here? I think we expect an error and we've got an EitherT, and the error itself is an Either, so we just want to get that.  And we only have Future because we might have a promo code involved.
I feel like the test itself is not clear as some of the expectation is here, can we just move this code into the test method?

Reply:
This is the same for all the other tests in this spec, they are all built up in the same way. I think it might be clearer if we did change them to move these definitions into the test itself but I don't want to do them all in this PR so I'm going to leave this as is.

Reply:
fair enough, leave it for another day 👍